app/tests/unit/components/Form/ProjectAnnualReportFormSummary.test.tsx

```diff
@@ -0,0 +1,220 @@
+                operation: "ARCHIVE",
+                changeStatus: "committed",
+                formChangeByPreviousFormChangeId: null,
+              },
+            },
+          ],
+        },
+      };
+    return result;
+  },
+};
+
+const defaultComponentProps = {
+  setValidatingForm: jest.fn(),
+  onSubmit: jest.fn(),
+};
+
+const componentTestingHelper =
+  new ComponentTestingHelper<ProjectAnnualReportFormSummaryQuery>({
+    component: ProjectAnnualReportFormSummary,
+    testQuery: testQuery,
+    compiledQuery: compiledProjectAnnualReportFormSummaryQuery,
+    getPropsFromTestQuery: (data) => ({
+      query: data.query,
+      projectRevision: data.query.projectRevision,
+    }),
+    defaultQueryResolver: mockQueryPayload,
+    defaultQueryVariables: {},
+    defaultComponentProps: defaultComponentProps,
+  });
+
+describe("The Project Annual Report Form Summary", () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    componentTestingHelper.reinit();
+  });
+
+  it("Only displays the data fields that have changed", () => {
+    componentTestingHelper.loadQuery();
+    componentTestingHelper.renderComponent();
+
+    expect(screen.getAllByText("Report Due Date")).toHaveLength(2);
+    expect(screen.getAllByText("Received Date (optional)")).toHaveLength(2);
+    expect(screen.getAllByText("General Comments (optional)")).toHaveLength(2);
+  });
+
+  it("Displays diffs of the the data fields that were updated", () => {
+    componentTestingHelper.loadQuery();
+    componentTestingHelper.renderComponent();
+
+    // First Node
+    expect(
+      screen.getByText("I am the old comment on Annual Report #1")
+    ).toBeInTheDocument();
+    expect(
+      screen.getByText("I am the updated comment on Annual Report #1")
+    ).toBeInTheDocument();
+    // updated due date
+    // TODO: update with the return of the ReadOnlyDateWidget
+    expect(
+      screen.getByText("2022-06-18T23:59:59.999-07:00")
+    ).toBeInTheDocument();
+    expect(
+      screen.getByText("2025-06-18T23:59:59.999-07:00")
+    ).toBeInTheDocument();
+    //updated received date
+    expect(
+      screen.getByText("2022-07-01T23:59:59.999-07:00")
+    ).toBeInTheDocument();
+    expect(
+      screen.getByText("2025-07-01T23:59:59.999-07:00")
+    ).toBeInTheDocument();
+
+    // Second Node
+    expect(
+      screen.queryByText("I am an unchanged comment on Annual Report #2")
+    ).not.toBeInTheDocument();
+    expect(
+      screen.queryByText("I am an unchanged due date on Annual Report #2")
+    ).not.toBeInTheDocument();
+    expect(
+      screen.queryByText("I am an unchanged received date on Annual Report #2")
+    ).not.toBeInTheDocument();
+
+    // Third Node
+    expect(screen.getByText("I am a new comment")).toBeInTheDocument();
+    expect(
+      screen.getByText("2021-01-01T23:59:59.999-07:00")
+    ).toBeInTheDocument();
+    expect(
+      screen.getByText("2022-01-01T23:59:59.999-07:00")
+    ).toBeInTheDocument();
```

Comment:
I'm wondering if we can use the actual date format rather than ISO date format but not sure it really matters. although worth mentioning that we have a function (`getLocaleFormattedDate`) that turns the ISO date to locale format on the `ReadOnlyDateWidget`.